x-pack/plugins/security_solution/public/common/components/ml/anomaly/use_anomalies_search.ts

```diff
@@ -129,21 +131,40 @@ function formatResultData(
     key: string;
     doc_count: number;
   }>,
-  notableAnomaliesJobs: SecurityJob[]
+  notableAnomaliesJobs: SecurityJob[],
+  spaceId: string | undefined
 ): AnomaliesCount[] {
-  const unsortedAnomalies: AnomaliesCount[] = NOTABLE_ANOMALIES_IDS.map((notableJobId) => {
-    const job = notableAnomaliesJobs.find(({ id }) => id === notableJobId);
-
-    const bucket = buckets.find(({ key }) => key === job?.id);
-    const hasUserName = has("entity.hits.hits[0]._source['user.name']", bucket);
-
-    return {
-      name: job?.customSettings?.security_app_display_name ?? notableJobId,
-      count: bucket?.doc_count ?? 0,
-      entity: hasUserName ? AnomalyEntity.User : AnomalyEntity.Host,
-      job,
-    };
+  const unsortedAnomalies: AnomaliesCount[] = NOTABLE_ANOMALIES_IDS.flatMap((notableJobId) => {
+    const jobs = notableAnomaliesJobs.filter(({ id }) => matchJobId(id, notableJobId, spaceId));
+
+    return jobs.map((job) => {
+      const bucket = buckets.find(({ key }) => key === job?.id);
+      const hasUserName = has("entity.hits.hits[0]._source['user.name']", bucket);
+
+      return {
+        name: job.customSettings?.security_app_display_name ?? notableJobId,
+        count: bucket?.doc_count ?? 0,
+        entity: hasUserName ? AnomalyEntity.User : AnomalyEntity.Host,
+        job,
+      };
+    });
   });
-
   return sortBy(['name'], unsortedAnomalies);
 }
+
+export const installedJobPrefix = (spaceId: string | undefined) => `${spaceId ?? 'default'}_`;
+
+export const uninstalledJobIdToInstalledJobId = (
+  moduleJobId: string,
+  spaceId: string | undefined
+) => `${installedJobPrefix(spaceId)}${moduleJobId}`;
+
+/**
+ * From version 8.8, jobs installed using security solution have the spaceId in their name.
+ * Jobs installed using security solution on versions older than 8.8 don't have the spaceId in their name.
```

Comment:
duplicated? 🤔

Reply:
I am deleting `Jobs installed using security solution on versions older than 8.8 don't have the spaceId in their name.
` because it is redundant and confusing